### ISSUE TITLE
Tetris N-Blox is broken

### ISSUE THREAD

**Issue by Electronik21** [ruffle-rs/ruffle#9242]
### Describe the bug

Tetris N-Blox doesn't work, when you should get a line clear it flashes and plays the sound, but the blocks stay there. It does it again every time you drop a piece and the lines still don't get cleared. Also, it doesn't detect game over and instead just places pieces at the top of the screen over and over again. No errors in js console

### Expected behavior

For lines to clear and the game to end correctly

### Affected platform

Desktop app

### Operating system

Windows 10

### Additional information

Here's the [swf file](https://github.com/ruffle-rs/ruffle/files/10471094/tetris-n-blox.zip)

This used to work, it worked in nightly-2023-01-08 and then broke in nightly-2023-01-10. I attached videos of running the game in [01-08](https://user-images.githubusercontent.com/85518515/213827726-6f1d9b98-24c1-4dc1-8929-f36c6a865470.mp4) and [01-10](https://user-images.githubusercontent.com/85518515/213827764-0d0a52b5-11e9-4be8-9db3-e74651f53d97.mp4). I've tested it in the windows app, the demo, and a self hosted game site and they all do the same thing. It doesn't give any warnings or errors in the javascript console either.

**Comment by Lord-McSweeney** [comment]
This has one `with (_root)` and two `with (_parent)`, it's possible that with scopes on those special properties are different.

**Comment by Toad06** [comment]
I added a couple of `trace`s in `/scripts/DefineSprite (2)/frame 7/DoAction`. This code is triggered when a line of blocks is destroyed.

```
x = Number(_parent._name) - y;
with(_root) {
   trace("y: " + y);
   var j = y; // <-- y doesn't seem to be read from the correct location here.
   trace("j: " + j);
   // (...)
}
```

Flash Player always traces the same value for `y` and `j` (which is 170 for the bottom line).
Ruffle traces a value that matches Flash for the `y` variable but the `j` one is completely unrelated (returns -4, then 6).

Regression intoduced in #8953.
